File: myFinance_improved_V2/lib/features/inventory_management/domain/entities/inventory_metadata.dart
// Domain Entity: Inventory Metadata
// Contains reference data for inventory management

class InventoryMetadata {
  final InventoryStats stats;
  final List<String> units;
  final List<Brand> brands;
  final Currency currency;
  final List<Category> categories;
  final StoreInfo storeInfo;
  final DateTime? lastUpdated;
  final List<ProductType> productTypes;
  final ValidationRules validationRules;
  final AllowCustomValues allowCustomValues;
  final List<StockStatusLevel> stockStatusLevels;

  InventoryMetadata({
    required this.stats,
    required this.units,
    required this.brands,
    required this.currency,
    required this.categories,
    required this.storeInfo,
    this.lastUpdated,
    required this.productTypes,
    required this.validationRules,
    required this.allowCustomValues,
    required this.stockStatusLevels,
  });
}

class InventoryStats {
  final int totalBrands;
  final int totalProducts;
  final int activeProducts;
  final int totalCategories;
  final int inactiveProducts;

  InventoryStats({
    required this.totalBrands,
    required this.totalProducts,
    required this.activeProducts,
    required this.totalCategories,
    required this.inactiveProducts,
  });
}

class Brand {
  final String id;
  final String name;
  final int? productCount;

  Brand({
    required this.id,
    required this.name,
    this.productCount,
  });
}

class Currency {
  final String? code;
  final String? name;
  final String? symbol;

  Currency({
    this.code,
    this.name,
    this.symbol,
  });

  String get displaySymbol => symbol ?? code ?? '';
}

class Category {
  final String id;
  final String name;
  final String? parentId;
  final int? productCount;
  final List<Category>? subcategories;

  Category({
    required this.id,
    required this.name,
    this.parentId,
    this.productCount,
    this.subcategories,
  });

  bool get hasSubcategories =>
      subcategories != null && subcategories!.isNotEmpty;
}

class StoreInfo {
  final String storeId;
  final String storeCode;
  final String storeName;

  StoreInfo({
    required this.storeId,
    required this.storeCode,
    required this.storeName,
  });
}

class ProductType {
  final String label;
  final String value;
  final String? description;
  final int productCount;

  ProductType({
    required this.label,
    required this.value,
    this.description,
    required this.productCount,
  });
}

class ValidationRules {
  final String? skuPattern;
  final String? barcodePattern;
  final bool maxStockRequired;
  final bool minPriceRequired;

  ValidationRules({
    this.skuPattern,
    this.barcodePattern,
    required this.maxStockRequired,
    required this.minPriceRequired,
  });
}

class AllowCustomValues {
  final bool units;
  final bool brands;
  final bool categories;

  AllowCustomValues({
    required this.units,
    required this.brands,
    required this.categories,
  });
}

class StockStatusLevel {
  final String icon;
  final String color;
  final String label;
  final String level;

  StockStatusLevel({
    required this.icon,
    required this.color,
    required this.label,
    required this.level,
  });

  static StockStatusLevel mock() => StockStatusLevel(
        icon: 'info',
        color: 'blue',
        label: 'Normal',
        level: 'normal',
      );
}

File: website/src/features/inventory_management/domain/entities/InventoryMetadata.ts
/**
 * Inventory Metadata Entities
 * Types for inventory dropdown data from get_inventory_metadata_v2 RPC
 */

export interface Category {
  category_id: string;
  category_name: string;
  description: string;
  is_active: boolean;
  product_count: number;
  can_have_subcategory: boolean;
}

export interface Brand {
  brand_id: string;
  brand_name: string;
  description: string;
  is_active: boolean;
  product_count: number;
}

export interface ProductType {
  value: string;
  label: string;
  description: string;
  product_count: number;
}

export interface Unit {
  value: string;
  label: string;
  symbol: string;
  product_count: number;
}

export interface Currency {
  code: string;
  name: string;
  symbol: string;
}

export interface StockStatusLevel {
  level: string;
  label: string;
  color: string;
  icon: string;
}

export interface ValidationRules {
  sku_pattern: string;
  barcode_pattern: string;
  max_stock_required: boolean;
  min_price_required: boolean;
}

export interface AllowCustomValues {
  units: boolean;
  brands: boolean;
  categories: boolean;
}

export interface StoreInfo {
  store_id: string | null;
  store_code: string | null;
  store_name: string | null;
}

/**
 * Attribute option for variant attributes (v2)
 */
export interface AttributeOption {
  option_id: string;
  option_value: string;
  sort_order: number;
  is_active: boolean;
}

/**
 * Attribute with options for multi-attribute variant support (v2)
 */
export interface Attribute {
  attribute_id: string;
  attribute_name: string;
  sort_order: number;
  is_active: boolean;
  option_count: number;
  options: AttributeOption[];
}

export interface Stats {
  total_categories: number;
  total_brands: number;
  total_products: number;
  active_products: number;
  inactive_products: number;
  // v2 fields
  total_attributes: number;
  total_options: number;
}

export interface InventoryMetadata {
  company_id: string;
  categories: Category[];
  brands: Brand[];
  product_types: ProductType[];
  units: Unit[];
  attributes: Attribute[];  // v2: Attributes with options for variant support
  currency: Currency;
  stock_status_levels: StockStatusLevel[];
  validation_rules: ValidationRules;
  allow_custom_values: AllowCustomValues;
  store_info: StoreInfo;
  stats: Stats;
  last_updated: string;
}

export interface InventoryMetadataResponse {
  success: boolean;
  data: InventoryMetadata;
}
